TableDependency.SqlClient/SqlTableDependency.cs

```diff
@@ -521,10 +521,29 @@ protected override bool CheckIfNeedsToCreateDatabaseObjects()
                 }
             }
 
-            if (allObjectAlreadyPresent.All(exist => !exist.Value)) return true;
-            if (allObjectAlreadyPresent.All(exist => exist.Value)) return false;
+            var valReturn = CheckValuesReturn(allObjectAlreadyPresent);
```

Comment:
talvez daria para fazer algo mais simples...

algo como allObjectAlreadyPresent.Values.Any(x => x == false). Ai dá pra rancar o método CheckValuesReturn

dá uma olhada nesse repl que fiz https://repl.it/MH5G/0